jbpm-test-coverage/src/test/java/org/jbpm/test/functional/log/ProcessInstanceLogArchiveTest.java

```diff
@@ -0,0 +1,125 @@
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.internal.task.api.InternalTaskService;
+
+/**
+ * TODO: Add deleteLogsByDate, deleteLogsByDateRange (however it is a bit redundant).
+ */
+public class ProcessInstanceLogArchiveTest extends JbpmTestCase {
+
+    private static final String HELLO_WORLD_PROCESS = "org/jbpm/test/functional/common/HumanTask.bpmn2";
+    private static final String HELLO_WORLD_PROCESS_ID = "org.jbpm.test.functional.common.HumanTask";
+
+
+    private JPAAuditLogService auditService;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        auditService = new JPAAuditLogService(getEmf());
+        auditService.clear();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        try {
+            auditService.clear();
+            auditService.dispose();
+        } finally {
+            super.tearDown();
+        }
+    }
+
+    @Test
+    public void checkEndDateArchived() {
+        KieSession kieSession = createKSession(HELLO_WORLD_PROCESS);
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("var", params);
+        startProcess(kieSession, HELLO_WORLD_PROCESS_ID, 2, params);
+
+        InternalTaskService taskService = (InternalTaskService) getRuntimeEngine().getTaskService();
+        taskService.getTasksAssignedAsPotentialOwnerByStatusByGroup("john", null, null).forEach(e -> {
+            taskService.start(e.getId(), "john");
+            taskService.complete(e.getId(), "john", Collections.emptyMap());
+        });
+
+        EntityManager em = getEmf().createEntityManager();
+
+        List<ProcessInstanceLog> pLogs = em.createQuery("SELECT o FROM ProcessInstanceLog o", ProcessInstanceLog.class).getResultList();
+        Assert.assertTrue(!pLogs.isEmpty());
+        for(ProcessInstanceLog log : pLogs) {
+            Assert.assertNotNull(log.getEnd());
+        }
+        List<Long> ids = pLogs.stream().map(ProcessInstanceLog::getId).collect(Collectors.toList());
+
+        List<NodeInstanceLog> nLogs = em.createQuery("SELECT o FROM NodeInstanceLog o", NodeInstanceLog.class).getResultList();
+        Assert.assertTrue(!nLogs.isEmpty());
+        for(NodeInstanceLog log : nLogs) {
+            Assert.assertNotNull(log.getEnd());
+        }
+
+        List<VariableInstanceLog> vLogs = em.createQuery("SELECT o FROM VariableInstanceLog o", VariableInstanceLog.class).getResultList();
+        Assert.assertTrue(!vLogs.isEmpty());
+        for(VariableInstanceLog log : vLogs) {
+            Assert.assertNotNull(log.getEnd());
+        }
+
+        List<TaskEventImpl> tLogs = em.createQuery("SELECT o FROM TaskEventImpl o WHERE o.processInstanceId IN (:var)", TaskEventImpl.class)
+                .setParameter("var", ids)
+                .getResultList();
+        Assert.assertTrue(!tLogs.isEmpty());
+        for(TaskEventImpl log : tLogs) {
+            System.out.println(log);
```

Comment:
I think it's not really needed to output this in console, if so, maybe it's better to use a logger instead

Reply:
removed